### DISCUSSION TITLE
How to interact with channels you manage?

### DISCUSSION THREAD

**Discussion by rchive1** [sns-sdks/python-youtube#166] [Q&A]
Let's say I want to insert a new video into a particular channel that I'm an Editor of. Going the Client, OAuth route, how do I do that? I see that `Client.videos.insert()` has parameters `managed_by_me` and `on_behalf_of_content_owner`, but I'm not sure what arguments to pass, exactly. If I just pass a channel ID string for `on_behalf_of_content_owner`, I get a `PyYouTubeException: YouTubeException(status_code=500,message=Internal error encountered.)` from `upload.next_chunk()`

Let me know if I need to give more information to make my question make sense.

Thanks!

**Comment by MerleLiuKun** [comment]
Refer official [docs](https://developers.google.com/youtube/v3/docs/videos/insert) 

you don't need parameter `on_behalf_of_content_owner`.

When you do autorize, you have already choosed the target channel to grant permission. after authorization, the api call will belong the channel.

**Reply by rchive1** [reply]
I think the problem may be that my YouTube account is a standard Google account, not a Workspace account, and I'm not a YouTube Partner nor do I have Content Manager and all that stuff. I lost track of all the names YouTube has for its different components and programs. I was just hoping upload a bunch of videos programmatically and to learn the API a little.

When authorizing on the OAuth page, I don't get any options to specify a channel, just my personal YouTube account.
